dsgrid/registry/registry_database.py

```diff
@@ -0,0 +1,416 @@
+
+    def get_latest(self, root_db_id):
+        cursor = self._client.aql.execute(
+            f"""
+            FOR v in 1
+                OUTBOUND "{root_db_id}"
+                GRAPH "{GRAPH}"
+                OPTIONS {{edgeCollections: ["{Edge.LATEST.value}"]}}
+                RETURN v
+        """,
+            count=True,
+        )
+        count = cursor.count()
+        assert count <= 1, f"{root_db_id=} {count=}"
+        if count == 0:
+            raise DSGValueNotRegistered(f"{root_db_id=} is not registered")
+        doc = cursor.next()
+        if doc is None:
+            raise DSGValueNotRegistered(f"{root_db_id=} is not registered")
+        return doc
+
+    def get_latest_version(self, root_db_id):
+        cursor = self._client.aql.execute(
+            f"""
+            FOR v, e, p in 1
+                OUTBOUND "{root_db_id}"
+                GRAPH "{GRAPH}"
+                OPTIONS {{edgeCollections: ["{Edge.LATEST.value}"]}}
+                RETURN p.vertices[1]._id
+        """,
+            count=True,
+        )
+        count = cursor.count()
+        assert count <= 1, f"{root_db_id=} {count=}"
+        if count == 0:
+            raise DSGValueNotRegistered(f"{root_db_id=} is not registered")
+        model_db_id = cursor.next()
+        if model_db_id is None:
+            raise DSGValueNotRegistered(f"{root_db_id=} is not registered")
+        return self.get_version(model_db_id)
+
+    def _get_by_version(self, db_id, version):
+        cursor = self._client.aql.execute(
+            f"""
+            FOR v, e, p in 0..{MAX_REGISTRY_VERSIONS}
+                OUTBOUND "{db_id}"
+                GRAPH "{GRAPH}"
+                OPTIONS {{edgeCollections: ["{Edge.UPDATED_TO.value}"]}}
+                FILTER e.version == "{version}"
+                RETURN p.vertices[-1]
+        """,
+            count=True,
+        )
+        count = cursor.count()
+        assert count <= 1, f"{db_id=} {count=}"
+        if count == 0:
+            raise DSGValueNotRegistered(f"{db_id=} {version=} is not registered")
+        return cursor.next()
+
+    def get_registration(self, db_id) -> RegistrationModel:
+        """Return the registration information for the id."""
+        cursor = self._client.aql.execute(
+            f"""
+            FOR v, e in 1
+                INBOUND "{db_id}"
+                GRAPH "{GRAPH}"
+                OPTIONS {{edgeCollections: ["{Edge.UPDATED_TO.value}"]}}
+                RETURN {{version: e.version, submitter: e.submitter, date: e.date,
+                        log_message: e.log_message}}
+        """,
+            count=True,
+        )
+        count = cursor.count()
+        assert count <= 1, f"{db_id=} {count=}"
+        if count == 0:
+            raise DSGValueNotRegistered(f"{db_id=} is not registered")
+
+        return RegistrationModel(**cursor.next())
+
+    def get_version(self, db_id):
+        """Return the dsgrid version of the database document.
```

Comment:
"dsgrid version" makes me think "software version", but this is really just the version of the registered item, right?

Reply:
Yes, I'm differentiating the dsgrid/user version from the database version (`_rev` field).